x-pack/plugins/canvas/public/state/middleware/es_persist.js

```diff
@@ -51,31 +51,44 @@ export const esPersistMiddleware = ({ getState }) => {
       return;
     }
 
-    // if the workpad changed, save it to elasticsearch
-    if (workpadChanged(curState, newState) || assetsChanged(curState, newState)) {
-      const persistedWorkpad = getWorkpadPersisted(getState());
-      return update(persistedWorkpad.id, persistedWorkpad).catch(err => {
-        const statusCode = err.response && err.response.status;
-        switch (statusCode) {
-          case 400:
-            return notify.error(err.response, {
-              title: `Couldn't save your changes to Elasticsearch`,
-            });
-          case 413:
-            return notify.error(
-              `The server gave a response that the workpad data was too large. This
+    const notifyError = err => {
+      const statusCode = err.response && err.response.status;
+      switch (statusCode) {
+        case 400:
+          return notify.error(err.response, {
+            title: `Couldn't save your changes to Elasticsearch`,
+          });
+        case 413:
+          return notify.error(
+            `The server gave a response that the workpad data was too large. This
               usually means uploaded image assets that are too large for Kibana or
               a proxy. Try removing some assets in the asset manager.`,
-              {
-                title: `Couldn't save your changes to Elasticsearch`,
-              }
-            );
-          default:
-            return notify.error(err, {
-              title: `Couldn't update workpad`,
-            });
-        }
-      });
+            {
+              title: `Couldn't save your changes to Elasticsearch`,
+            }
+          );
+        default:
+          return notify.error(err, {
+            title: `Couldn't update workpad`,
+          });
+      }
+    };
+
+    const changedWorkpad = workpadChanged(curState, newState);
+    const changedAssets = assetsChanged(curState, newState);
+
+    if (changedWorkpad && changedAssets) {
```

Comment:
Interestingly, I can never hit this scenario. Which makes sense, all the state updates are sync and happen in order, and assets and workpads all have their own distinct actions.